src/main/java/com/mraof/minestuck/item/weapon/InventoryTickEffect.java

```diff
@@ -48,4 +50,16 @@ static InventoryTickEffect passiveAspectEffect(EnumAspect aspect, Supplier<Effec
 			}
 		};
 	}
+	
+	InventoryTickEffect SURF_N_TURF = (stack, worldIn, entityIn, itemSlot, isSelected) -> {
+		if(isSelected && entityIn instanceof ServerPlayerEntity)
+		{
+			ServerPlayerEntity serverPlayerEntity = (ServerPlayerEntity) entityIn;
+			ItemStack otherStack = !serverPlayerEntity.getHeldItemMainhand().isItemEqual(stack) ? serverPlayerEntity.getHeldItemMainhand() : serverPlayerEntity.getHeldItemOffhand(); //it gives the stack in the hand that doesnt hold the tick weapon
+			if(ItemStack.areItemsEqual(MSItems.PRISMARINE_BASHER.getDefaultInstance(), otherStack))
+			{
+				MSCriteriaTriggers.SURF_N_TURF.trigger(serverPlayerEntity);
+			}
+		}
+	};
```

Comment:
This method works, but I am opening to optimization or alternatives. I tried picking what is likely the less popular weapon for the tick effect so that it is less likely to cause performance issues

Reply:
Since you're comparing an item stack and an item and not two itemstacks, you could compare the items directly (`MsItems.PRISMARINE_BASHER == otherStack.getItem()`)
Next, it looks like `isSelected` is unreliable for items not in the main inventory, so I'd suggest you check directly if it's in the right place (with `serverPlayerEntity.getHeldItemMainhand() == stack` and `serverPlayerEntity.getItemStackFromSlot(EquipmentSlotType.OFFHAND) == stack`)
For this specific purpose, its fine to use `==` instead of `stack.isItemEqual(stack2)`, since we're looking for the exact stack, and not just an equivalent stack.